src/elm/Main.elm

```diff
@@ -763,8 +763,11 @@ update msg model =
 
                             else
                                 Cmd.none
+
+                        decodedLog =
```

Comment:
this is the meat of the enhancement, decode the log and overwrite the data, no need to do this in view/render and we shouldnt